dotcom-rendering/src/web/components/EnhancePinnedPost.importable.tsx

```diff
@@ -78,11 +82,16 @@ export const EnhancePinnedPost = () => {
 
 	const pinnedPostTiming = useRef<ReturnType<typeof initPerf>>();
 
-	const contentFitsContainer =
-		pinnedPostContent &&
-		pinnedPostContent.scrollHeight <= pinnedPostContent.clientHeight;
+	/**
+	 * Timeout allows embeds to load before calculating content height
+	 */
+	useEffect(() => {
```

Comment:
@oliverlloyd tells me this bug may have been introduced a couple of weeks back when they started to defer enhancing tweets until they're in view. So perhaps it may be better to change that logic and not defer enhancing the tweet if it's inside a pinned post? This way we can avoid a timeout hopefully.

Reply:
This is a fiddly one!

I did initially thought that this could have been related to the work we did to defer tweet enhancement but after looking further I don't think this was the issue. I think we've always had this race condition. Essentially, we need to be able to trigger the height recalculation based on the DOM being updated. Luckily, I had a meeting with @mxdvl earlier where he mentioned [mutation observers](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) and I have a feeling this is what we need here.

I tried the code below locally and it worked for the use case that this PR is addressing but I didn't test it for all other situations

```ts
	const checkContentHeight = () => {
		const contentFitsContainer =
			pinnedPostContent &&
			pinnedPostContent.scrollHeight <= pinnedPostContent.clientHeight;
		if (contentFitsContainer) hideShowMore();
	};

	useEffect(() => {
		if (!pinnedPost) return;
		const observer = new MutationObserver(checkContentHeight);
		observer.observe(pinnedPost);

		return () => observer.disconnect();
	}, []);
```

Reply:
Thanks @oliverlloyd ! Much appreciate you looking at it. This works beautifully when a tweet starts above the height cut off for a pinned post.

However when tweet is initially hidden, after adding the above suggestion the tweet no longer enhances. Do you have any idea what might be causing that?

An example live blog / pinned post is here:
http://localhost:3030/Article?url=https://www.theguardian.com/politics/live/2022/may/17/northern-ireland-protocol-brexit-liz-truss-boris-johnson-uk-politics-latest 

Or here:
http://localhost:3030/Article?url=https://www.theguardian.com/business/live/2022/may/13/markets-volatile-ftse-wall-street-oil-bitcoin-crypto-recession-worries-business-live

Reply:
Great spot on the config fix 👍